### PR TITLE
MP/ Split customer/seller signup forms with new routes

### PR DESCRIPTION
**Pull request by MPucheta**
There is some "sharing" that can be abstracted, but I really feel like "Customer" and "Seller" are two different actors that will evolve differently, hence they shouldn't share things.

Also this PR comes with `Select` and `Option`, the same that we had in COPA, but with some tweaks (now it shows the error if the validation fails, for example

image

### CODE REVIEW

**Review by roke77 — APPROVED** [review]

**Review by eigiachero — APPROVED** [review]

File: src/modules/Auth/components/SignUpSellerForm/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+}
+
+.button {
+  margin-top: 28px;
+}
+
+.formInline {
+  display: grid;
+  grid-template-columns: repeat(2,1fr);
```
Comment (nahudalla) [inline src/modules/Auth/components/SignUpSellerForm/styles.module.scss]:
```suggestion
  grid-template-columns: repeat(2, 1fr);
```

File: src/modules/Auth/components/SignUpSellerForm/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  margin-top: 16px;
+  font-size: 12px;
+  line-height: 18px;
+  color: $gray;
+}
+
+.news {
+  text-align: center;
+  margin-top: 20px;
+  color:$black;
```
Comment (nahudalla) [inline src/modules/Auth/components/SignUpSellerForm/styles.module.scss]:
```suggestion
  color: $black;
```

File: src/modules/Auth/components/SignUpSellerForm/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  }
+}
+
+.labels {
+  width: 45px;
+  height: 18px;
+  margin-bottom: 16px;
+  font-size: 18px;
+  line-height: 18px;
+  color:$black;
```
Comment (nahudalla) [inline src/modules/Auth/components/SignUpSellerForm/styles.module.scss]:
```suggestion
  color: $black;
```

File: src/modules/Auth/pages/SignUpCustomer/index.js [hunk 4]
```diff
@@ -2,26 +2,26 @@ import React, { useCallback, useContext } from 'react'
...
   const { signup, loading } = useSignup()
   const history = useHistory()
 
-  const handleSubmit = useCallback(async ({ email, password, firstName, lastName }) => {
-    const { token } = await signup({ email, password, name: `${firstName} ${lastName}` })
+  const handleSubmit = useCallback(async ({ email, password, firstName, lastName, ...restValues }) => {
+    const { token } = await signup({ name: `${firstName} ${lastName}`, ...restValues })
 
-    await logFormSubmit('signup_form')
+    await logFormSubmit('customer_signup_form')
```
Comment (nahudalla) [inline src/modules/Auth/pages/SignUpCustomer/index.js]: **Caution!** This isn't just another name to change lightly. This is used in Google Analytics. Maybe we need to think about it more thoroughly.

File: src/modules/MainApp/components/Select/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types'
+import { Field, useField } from 'formik'
+import classnames from 'classnames'
+import AngleDown from '~/src/vectors/angle-down.svg'
+
+import Option from '../Option'
+
+import classes from './styles.module.scss'
+
+const Select = ({ className, children, ...props }) => {
```
Comment (nahudalla) [inline src/modules/MainApp/components/Select/index.js]: This Select is not correctly implemented... Maybe it was an old implementation, but we should consider using a library such as `react-select` for these cases. The main problem is accessibility. For example, the _focus_ of an element is not just a visual matter, it's a functional one. I see that we're applying "focus" styles conditionally, but the input is never really focused by the browser.

**Review by nahudalla — CHANGES_REQUESTED** [review]

File: src/modules/MainApp/components/Select/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  letter-spacing: 0.01em;
+  color: $error;
+}
+
+.error {
+  border: 0.9px solid $error;
+  border-radius: 4px;
+}
+
+:global .react-select__control {
```
Comment (nahudalla) [inline src/modules/MainApp/components/Select/styles.module.scss]: Don't use global CSS classes. You can apply it specifically for this component with the following selector:
```scss
.selectContainer {
  [class^='react-select__control'] {
    // ... your styles
  }
}
```

File: src/modules/MainApp/components/Select/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  padding-left: 16px;
+  height: 45px;
+  border-radius: 6px;
+  appearance: none;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+  cursor: pointer;
+}
+
+:global .react-select__value-container {
```
Comment (nahudalla) [inline src/modules/MainApp/components/Select/styles.module.scss]: Same here.

**Review by nahudalla — CHANGES_REQUESTED** [review]

**Review by nahudalla — APPROVED** [review]

**Pull request merged**